src-v2/Not/Not/Structures/SelectListModel.cs

```diff
@@ -4,6 +4,12 @@ namespace Not.Structures;
 
 public class SelectListModel
 {
+    public static IEnumerable<SelectListModel> SelectList<T>()
+    {
+        var enumValues = Enum.GetValues(typeof(T)).Cast<Enum>();
+        var selectItems = enumValues.Select(s => new SelectListModel(s.ToString(), s.GetDescription()));
+        return selectItems;
+    }
```

Comment:
What if `T` is not an enum. While constraining a generic param to enum type isn't entirely possible on compile time, there are 3 major things that we can do to prevent errors:
1. Constraint T to struct:
```
public static IEnumerable<SelectListModel> SelectList<T>()
    where T : struct
```
This prevents the method from compiling successfully if it's attempted to be used with reference types, aka classes. 

2. Verify that the type is enum - using `Type.IsEnum` property you can determine whether or not T is an Enum type and throw meaningful error if it is not. This is an OK solution, since typeof(T) isn't an expensive reflection operation. 

3. Most notably - you can indicate that T is enum in the method name.